go/api/response_types.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+type Result[T any] struct {
+	val   T
+	isNil bool
```

Comment:
Sorry @janhavigupta007 I know this is already merged, but could you please help me to understand why it is necessary to have a `isNil` boolean field here? Thank you!